src/webgpu/shader/validation/expression/access/vector.spec.ts

```diff
@@ -0,0 +1,203 @@
+    ok: (width: number) => width > 3,
+  },
+  const_expr_2_via_struct: {
+    wgsl: 'let r : T = v[S(2).i];',
+    ok: (width: number) => width > 2,
+  },
+  const_expr_3_via_struct: {
+    wgsl: 'let r : T = v[S(3).i];',
+    ok: (width: number) => width > 3,
+  },
+
+  // single element convenience name accesses
+  x: { wgsl: 'let r : T = v.x;', ok: true },
+  y: { wgsl: 'let r : T = v.y;', ok: true },
+  z: { wgsl: 'let r : T = v.z;', ok: (width: number) => width > 2 },
+  w: { wgsl: 'let r : T = v.w;', ok: (width: number) => width > 3 },
+  r: { wgsl: 'let r : T = v.r;', ok: true },
+  g: { wgsl: 'let r : T = v.g;', ok: true },
+  b: { wgsl: 'let r : T = v.b;', ok: (width: number) => width > 2 },
+  a: { wgsl: 'let r : T = v.a;', ok: (width: number) => width > 3 },
+
+  // swizzles
+  xy: { wgsl: 'let r : vec2<T> = v.xy;', ok: true },
+  yx: { wgsl: 'let r : vec2<T> = v.yx;', ok: true },
+  xyx: { wgsl: 'let r : vec3<T> = v.xyx;', ok: true },
+  xyz: { wgsl: 'let r : vec3<T> = v.xyz;', ok: (width: number) => width > 2 },
+  zyx: { wgsl: 'let r : vec3<T> = v.zyx;', ok: (width: number) => width > 2 },
+  xyxy: { wgsl: 'let r : vec4<T> = v.xyxy;', ok: true },
+  xyxz: { wgsl: 'let r : vec4<T> = v.xyxz;', ok: (width: number) => width > 2 },
+  xyzw: { wgsl: 'let r : vec4<T> = v.xyzw;', ok: (width: number) => width > 3 },
+  yxwz: { wgsl: 'let r : vec4<T> = v.yxwz;', ok: (width: number) => width > 3 },
+  rg: { wgsl: 'let r : vec2<T> = v.rg;', ok: true },
+  gr: { wgsl: 'let r : vec2<T> = v.gr;', ok: true },
+  rgg: { wgsl: 'let r : vec3<T> = v.rgg;', ok: true },
+  rgb: { wgsl: 'let r : vec3<T> = v.rgb;', ok: (width: number) => width > 2 },
+  grb: { wgsl: 'let r : vec3<T> = v.grb;', ok: (width: number) => width > 2 },
+  rgbr: { wgsl: 'let r : vec4<T> = v.rgbr;', ok: (width: number) => width > 2 },
+  rgba: { wgsl: 'let r : vec4<T> = v.rgba;', ok: (width: number) => width > 3 },
+  gbra: { wgsl: 'let r : vec4<T> = v.gbra;', ok: (width: number) => width > 3 },
+
+  // swizzle chains
+  xy_yx: { wgsl: 'let r : vec2<T> = v.xy.yx;', ok: true },
+  xyx_xxy: { wgsl: 'let r : vec3<T> = v.xyx.xxy;', ok: true },
+  xyz_zyx: { wgsl: 'let r : vec3<T> = v.xyz.zyx;', ok: (width: number) => width > 2 },
+  xyxy_rrgg: { wgsl: 'let r : vec4<T> = v.xyxy.rrgg;', ok: true },
+  rbrg_xyzw: { wgsl: 'let r : vec4<T> = v.rbrg.xyzw;', ok: (width: number) => width > 2 },
+  xyxz_rbg_yx: { wgsl: 'let r : vec2<T> = v.xyxz.rbg.yx;', ok: (width: number) => width > 2 },
+  wxyz_bga_xy: { wgsl: 'let r : vec2<T> = v.wxyz.bga.xy;', ok: (width: number) => width > 3 },
+
+  // error: invalid convenience letterings
+  xq: { wgsl: 'let r : vec2<T> = v.xq;', ok: false },
+  py: { wgsl: 'let r : vec2<T> = v.py;', ok: false },
+
+  // error: mixed convenience letterings
+  xg: { wgsl: 'let r : vec2<T> = v.xg;', ok: false },
+  ryb: { wgsl: 'let r : vec3<T> = v.ryb;', ok: false },
+  xgza: { wgsl: 'let r : vec4<T> = v.xgza;', ok: false },
```

Comment:
Maybe add one for too many components? e.g. `v.xxxxx`

Reply:
Done